### ISSUE TITLE
Why

### ISSUE THREAD

**Issue by Kdecem** [dmlc/mxnet#5444]
A: I got a poor ACC and it still betwwen 0 ---0.005,

B: when i am running, i got a warning " mxnet.model.FeedForward has been deprecated. Please use mxnet.mod.Module instead" and "Calling initializer with init(str, NDArray) has been deprecated.please use init(mx.init.InitDesc(...), NDArray) instead.",these will influnse my work?

", i don't know how to use "mx.mod.module"
C: I found use im2rec with recursive ture or false all lead the same label,and when i use shuffle false,it also can't work, i got a randomize file. 
**result info:**

2017-03-17 09:21:13,137 Node[0] Epoch[18] Batch [17300]	Speed: 258.90 samples/sec	Train-top_k_accuracy_10=0.001250
2017-03-17 09:21:13,137 Node[0] Epoch[18] Batch [17300]	Speed: 258.90 samples/sec	Train-top_k_accuracy_20=0.003750
2017-03-17 09:21:16,237 Node[0] Epoch[18] Batch [17350]	Speed: 258.04 samples/sec	Train-accuracy=0.000000
2017-03-17 09:21:16,237 Node[0] Epoch[18] Batch [17350]	Speed: 258.04 samples/sec	Train-top_k_accuracy_5=0.001250
2017-03-17 09:21:16,238 Node[0] Epoch[18] Batch [17350]	Speed: 258.04 samples/sec	Train-top_k_accuracy_10=0.001250
2017-03-17 09:21:16,238 Node[0] Epoch[18] Batch [17350]	Speed: 258.04 samples/sec	Train-top_k_accuracy_20=0.002500
2017-03-17 09:21:19,317 Node[0] Epoch[18] Batch [17400]	Speed: 259.79 samples/sec	Train-accuracy=0.000000
2017-03-17 09:21:19,317 Node[0] Epoch[18] Batch [17400]	Speed: 259.79 samples/sec	Train-top_k_accuracy_5=0.000000
2017-03-17 09:21:19,317 Node[0] Epoch[18] Batch [17400]	Speed: 259.79 samples/sec	Train-top_k_accuracy_10=0.000000
2017-03-17 09:21:19,317 Node[0] Epoch[18] Batch [17400]	Speed: 259.79 samples/sec	Train-top_k_accuracy_20=0.001250
2017-03-17 09:21:22,382 Node[0] Epoch[18] Batch [17450]	Speed: 261.07 samples/sec	Train-accuracy=0.000000
2017-03-17 09:21:22,382 Node[0] Epoch[18] Batch [17450]	Speed: 261.07 samples/sec	Train-top_k_accuracy_5=0.000000
2017-03-17 09:21:22,382 Node[0] Epoch[18] Batch [17450]	Speed: 261.07 samples/sec	Train-top_k_accuracy_10=0.000000

**this is my list file :**
198956	2502.000000	2077/00000535.jpg
26282	393.000000	10350/028.jpg
203015	2571.000000	2139/168.jpg
246952	3195.000000	2700/066.jpg
674824	9463.000000	8342/027.jpg
181413	2256.000000	1856/004.jpg
512786	7096.000000	6211/00000848.jpg
145517	1730.000000	1382/00000525.jpg
738795	10317.000000	9110/008.jpg
162113	1966.000000	1595/046.jpg

**mycode** 
from common import find_mxnet
import mxnet as mx
import argparse
import os, sys
from common import train_model
def googlenet():
	def ConvFactory(data, num_filter, kernel, stride=(1,1), pad=(0, 0), name=None, suffix=''):
   		 conv = mx.symbol.Convolution(data=data, num_filter=num_filter, kernel=kernel, 				stride=stride,pad=pad, name='conv_%s%s' %(name, suffix))
   
    		 act = mx.symbol.Activation(data=conv, act_type='relu', name='relu_%s%s' %(name, suffix))
    		 return act

	def InceptionFactory(data, num_1x1, num_3x3red, num_3x3, num_d5x5red, num_d5x5, pool, proj, name):
    # 1x1
   	 c1x1 = ConvFactory(data=data, num_filter=num_1x1, kernel=(1, 1), name=('%s_1x1' % name))
  
    # 3x3 reduce + 3x3
   	 c3x3r = ConvFactory(data=data, num_filter=num_3x3red, kernel=(1, 1), name=('%s_3x3' % name), 			suffix='_reduce')
   	 c3x3 = ConvFactory(data=c3x3r, num_filter=num_3x3, kernel=(3, 3), pad=(1, 1), name=('%s_3x3' % name))
    # double 3x3 reduce + double 3x3
   	 cd5x5r = ConvFactory(data=data, num_filter=num_d5x5red, kernel=(1, 1), name=('%s_5x5' % name), 		suffix='_reduce')
   	 cd5x5 = ConvFactory(data=cd5x5r, num_filter=num_d5x5, kernel=(5, 5), pad=(2, 2), name=('%s_5x5' % 			name))
    # pool + proj
  	 pooling = mx.symbol.Pooling(data=data, kernel=(3, 3), stride=(1, 1), pad=(1, 1 ), pool_type=pool, 			name=('%s_pool_%s_pool' % (pool, name)))
   	 cproj = ConvFactory(data=pooling, num_filter=proj, kernel=(1, 1), name=('%s_proj' %  name))
    # concat
	 concat = mx.symbol.Concat(*[c1x1, c3x3, cd5x5, cproj], name='ch_concat_%s_chconcat' % name)
   	 return concat
#######################
	def InceptionFactory2(data, num_3x3red, num_3x3, num_d5x5red, num_d5x5, pool, proj, name):
 
    # 3x3 reduce + 3x3
   	 c3x3r = ConvFactory(data=data, num_filter=num_3x3red, kernel=(1, 1), name=('%s_3x3' % name), 			suffix='_reduce')
	
  	 c3x3 = ConvFactory(data=c3x3r, num_filter=num_3x3, kernel=(3, 3), stride=(2, 2), pad=(1, 1), name=('%s_3x3' % name))
    # double 3x3 reduce + double 3x3
    	 cd5x5r = ConvFactory(data=data, num_filter=num_d5x5red, kernel=(1, 1), name=('%s_5x5' % name), suffix='_reduce')
   	 cd5x5 = ConvFactory(data=cd5x5r, num_filter=num_d5x5, kernel=(5, 5),stride=(2, 2),  pad=(2, 2), name=('%s_5x5' % name))
    # pool + proj
   	 pooling = mx.symbol.Pooling(data=data, kernel=(3, 3), stride=(2, 2), pad=(1, 1 ), pool_type=pool, name=('%s_pool_%s_pool' % (pool, name)))
    
    # concat
   	 concat = mx.symbol.Concat(*[c3x3, cd5x5, pooling], name='ch_concat_%s_chconcat' % name)
   	 return concat

	
 	data  = mx.sym.Variable("data")
 	conv1 = ConvFactory(data, 64, kernel=(7, 7), stride=(2,2), pad=(3, 3), name="conv1")

 	pool1 = mx.sym.Pooling(conv1, kernel=(3, 3), stride=(2, 2), pool_type="max")
 	conv2 = ConvFactory(pool1, 64, kernel=(1, 1), stride=(1,1), name="conv2")
 	conv3 = ConvFactory(conv2, 192, kernel=(3, 3), stride=(1, 1), pad=(1,1), name="conv3")
 	pool3 = mx.sym.Pooling(conv3, kernel=(3, 3), stride=(2, 2), pool_type="max")

 	in3a = InceptionFactory(pool3, 64, 96, 128, 16, 32, "max", 32, name="in3a")
 	in3b = InceptionFactory(in3a, 64, 96, 128, 32, 64, "max", 64, name="in3b")
#######################
 	in3c = InceptionFactory2(in3b, 128, 256, 32, 64, "max", 64, name="in3c")
########################
    #pool4 = mx.sym.Pooling(in3c, kernel=(3, 3), stride=(2, 2),pad=(1, 1), pool_type="max")
 	in4a = InceptionFactory(in3c, 256, 96, 192, 32, 64, "max", 64, name="in4a")
	in4b = InceptionFactory(in4a, 224, 112, 224, 32, 64, "max", 64, name="in4b")
 	in4c = InceptionFactory(in4b, 192, 128, 256, 32, 64, "max", 64, name="in4c")
	in4d = InceptionFactory(in4c, 160, 144, 288, 32, 64, "max", 64, name="in4d")

 	in4e = InceptionFactory2(in4d, 160, 256, 64, 128, "max", 128, name="in4e")
   # pool5 = mx.sym.Pooling(in4e, kernel=(3, 3), stride=(2, 2), pool_type="max")
 	in5a = InceptionFactory(in4e,384, 192, 384, 48, 128, "max", 128, name="in5a")
 	in5b = InceptionFactory(in5a, 384, 192, 384, 48, 128, "max", 128, name="in5b")
 	pool6 = mx.sym.Pooling(in5b, kernel=(7, 7), stride=(1,1), pool_type="avg")
 	flatten = mx.sym.Flatten(data=pool6)
 	fc1 = mx.sym.FullyConnected(data=flatten, num_hidden=1000)

 	softmax = mx.symbol.SoftmaxOutput(data=fc1, name='softmax')
   		 
 	return softmax

def get_iterator(args, kv):
    data_shape = (3, 224, 224)
    train = mx.io.ImageRecordIter(
        path_imgrec = "/media/pixel/ssd512G/tran.rec",
        mean_img ="/media/pixel/ssd512G/mean2.bin",
	data_name ='data',
	shuffle = True,
        data_shape  = data_shape,
        batch_size  = args.batch_size,
        rand_crop   = True,
        rand_mirror = True,
        num_parts   = kv.num_workers,
        part_index  = kv.rank)

    val = mx.io.ImageRecordIter(
        path_imgrec = "/media/pixel/ssd512G/tran0.rec",
        mean_img ="/media/pixel/ssd512G/mean21.bin",
	data_name='data',
	shuffle = False,
        rand_crop   = False,
        rand_mirror = False,
        data_shape  = data_shape,
        batch_size  = args.batch_size,
        num_parts   = kv.num_workers,
        part_index  = kv.rank)

    return (train, val)

def parse_args():
    parser = argparse.ArgumentParser(description='train an image classifer on mnist')
    parser.add_argument('--network', type=str, default='googlenet',
                        help = 'the cnn to use')
    parser.add_argument('--data-dir', type=str, default='/media/pixel/ssd512G/',
                        help='the input data directory')
    parser.add_argument('--gpus', type=str,default='0,1',
                        help='the gpus will be used, e.g "0,1,2,3"')
    parser.add_argument('--num-examples', type=int, default=5359,
                        help='the number of training examples')
    parser.add_argument('--batch-size', type=int, default=16,
                        help='the batch size')
    parser.add_argument('--lr', type=float, default=.01,
                        help='the initial learning rate')
    parser.add_argument('--model-prefix', type=str,
                        help='the prefix of the model to load/save')
    parser.add_argument('--save-model-prefix', type=str,
                        help='the prefix of the model to save')
    parser.add_argument('--num-epochs', type=int, default=2000,
                        help='the number of training epochs')
    parser.add_argument('--load-epoch', type=int,
                        help="load the model on an epoch using the model-prefix")
    parser.add_argument('--kv-store', type=str, default='local',
                        help='the kvstore type')
    parser.add_argument('--lr-factor', type=float, default=0.5,
                        help='times the lr with a factor for every lr-factor-epoch epoch')
    parser.add_argument('--lr-factor-epoch', type=float, default=1,
                        help='the number of epoch to factor the lr, could be .5')
    return parser.parse_args()

if __name__ == '__main__':
    args = parse_args()
    
    net =googlenet()
    # train
    train_model.fit(args, net, get_iterator)

## Environment info
Operating System: Ubuntu 

Package used (Python/R/Scala/Julia): Python

MXNet version: 0.9.3

**Comment by Kdecem** [comment]
I have used lr==0.01 lr=0.05 lr=0.001 batchsize =32 or 64 , i think  my fault is not about my para.

**Comment by Kdecem** [comment]
now i  think i got the right result with im2rec.py in 0.8.0,  the wrong all beacasue of my awful data

**Comment by Kdecem** [comment]
Closed

**Comment by ysh329** [comment]
@Kdecem Please redefine the problem according to the description, which may help more people like you, thanks :)
